Administrative state eventually is not persistent due to save method being executed in a thread

We've observed that end-to-end tests eventually fail with an administrative state not being saved to the storehouse. One example is shown below:
```
___ TestE2ETopology.test_080_disabling_all_interfaces_on_a_switch_persistent ___

self = <tests.test_e2e_05_topology.TestE2ETopology instance at 0x7fdec2349cb0>

   def test_080_disabling_all_interfaces_on_a_switch_persistent(self):
       """
       Test /api/kytos/topology/v3/interfaces/{interface_id}/disable on POST
       supported by:
           /api/kytos/topology/v3/interfaces/switch/{dpid}/enable on POST
           and
           /api/kytos/topology/v3/switches on GET
       """

       switch_id = "00:00:00:00:00:00:00:01"

       # Enabling all the interfaces
       api_url = KYTOS_API + '/topology/v3/interfaces/switch/%s/enable' % switch_id
       response = requests.post(api_url)
       assert response.status_code == 200

       self.restart()

       # Make sure all the interfaces belonging to the target switch are enabled
       api_url = KYTOS_API + '/topology/v3/switches'
       response = requests.get(api_url)
       data = response.json()

       for interface in data['switches'][switch_id]['interfaces']:
         assert data['switches'][switch_id]['interfaces'][interface]['enabled'] is True
E           assert False is True

tests/test_e2e_05_topology.py:332: AssertionError
```

I've repeated this specific test multiple times trying to reproduce this behavior. After 150 executions, only one fail was reported.

Investigating the failure, I've noticed that the Topology Napp is persisting the user request through a call to `save_status_on_storehouse` (https://github.com/kytos-ng/topology/blob/master/main.py#L351). This method includes the `@listen` decorator, which means it will be run in a thread. Since it is executed in a thread, the rest endpoint will return to the user right away with a 200 status code, but there is no guarantee that the persistent will be also executed right away. In particular, in the test above, after a successful 200 status code, the end-to-end test will request a shutdown for Kytos. Which means that if the `save_status_on_storehouse` thread don't gets started before the shutdown request gets received, the user request will be lost.

From the Kytos logs, we see exactly this behavior (faulty behavior):
```
Dec  2 18:02:17 31861dd23a4c 2021-12-02 18:02:17,768 werkzeug:INFO _internal:113:  127.0.0.1 - - [02/Dec/2021 18:02:17] "#033[37mPOST /api/kytos/topology/v3/interfaces/switch/00:00:00:00:00:00:00:01/enable HTTP/1.1#033[0m" 200 -
Dec  2 18:02:17 31861dd23a4c 2021-12-02 18:02:17,799 kytos.core.controller:INFO kytosd:124:  Stopping Kytos controller...
Dec  2 18:02:17 31861dd23a4c 2021-12-02 18:02:17,799 kytos.core.controller:INFO controller:404:  Stopping Kytos
Dec  2 18:02:17 31861dd23a4c 2021-12-02 18:02:17,799 kytos.core.buffers:INFO buffers:157:  Stop signal received by Kytos buffers.
Dec  2 18:02:17 31861dd23a4c 2021-12-02 18:02:17,800 kytos.core.buffers:INFO buffers:158:  Sending KytosShutdownEvent to all apps.
Dec  2 18:02:17 31861dd23a4c 2021-12-02 18:02:17,800 kytos.core.buffers:INFO buffers:46:  [buffer: raw_event] Stop mode enabled. Rejecting new events.
Dec  2 18:02:17 31861dd23a4c 2021-12-02 18:02:17,800 kytos.core.buffers:INFO buffers:46:  [buffer: msg_in_event] Stop mode enabled. Rejecting new events.
Dec  2 18:02:17 31861dd23a4c 2021-12-02 18:02:17,800 kytos.core.buffers:INFO buffers:46:  [buffer: msg_out_event] Stop mode enabled. Rejecting new events.
Dec  2 18:02:17 31861dd23a4c 2021-12-02 18:02:17,801 kytos.core.buffers:INFO buffers:46:  [buffer: app_event] Stop mode enabled. Rejecting new events.
```

As can be seen above, the user request (`POST .* topology/v3/interfaces/switch/xxx/enable`) was received and processed at `18:02:17,768`, returning a 200 status code. Then, 31 milliseconds later (at 2021-12-02 18:02:17,799) kytos receives the shutdown request (as for the `self.restart()` end-to-end test request), followed by all buffers entering in a `Stop mode` and `Rejecting new events ` (in this case, it will reject the event for persist data on the storehouse - meaning the user request will not be persisted).

Just for comparison, below is the log for a non-faulty behavior:
```
Dec  2 18:03:26 31861dd23a4c 2021-12-02 18:03:26,775 werkzeug:INFO _internal:113:  127.0.0.1 - - [02/Dec/2021 18:03:26] "#033[37mPOST /api/kytos/topology/v3/interfaces/switch/00:00:00:00:00:00:00:01/enable HTTP/1.1#033[0m" 200 -
Dec  2 18:03:26 31861dd23a4c 2021-12-02 18:03:26,784 kytos.napps.kytos/storehouse:INFO fs:83:  _load_from_file done /var/tmp/kytos/storehouse/kytos.topology.status/bb5913feb47845b0884586645b4d98a4
Dec  2 18:03:26 31861dd23a4c 2021-12-02 18:03:26,788 kytos.napps.kytos/storehouse:INFO fs:73:  _write_to_file done /var/tmp/kytos/storehouse/kytos.topology.status/bb5913feb47845b0884586645b4d98a4
Dec  2 18:03:26 31861dd23a4c 2021-12-02 18:03:26,791 kytos.napps.kytos/topology:INFO storehouse:114:  Network administrative status saved in kytos.topology.status.bb5913feb47845b0884586645b4d98a4
Dec  2 18:03:26 31861dd23a4c 2021-12-02 18:03:26,796 kytos.core.controller:INFO kytosd:124:  Stopping Kytos controller...
Dec  2 18:03:26 31861dd23a4c 2021-12-02 18:03:26,796 kytos.core.controller:INFO controller:404:  Stopping Kytos
Dec  2 18:03:26 31861dd23a4c 2021-12-02 18:03:26,797 kytos.core.buffers:INFO buffers:157:  Stop signal received by Kytos buffers.
Dec  2 18:03:26 31861dd23a4c 2021-12-02 18:03:26,797 kytos.core.buffers:INFO buffers:158:  Sending KytosShutdownEvent to all apps.
Dec  2 18:03:26 31861dd23a4c 2021-12-02 18:03:26,797 kytos.core.buffers:INFO buffers:46:  [buffer: raw_event] Stop mode enabled. Rejecting new events.
Dec  2 18:03:26 31861dd23a4c 2021-12-02 18:03:26,797 kytos.core.buffers:INFO buffers:46:  [buffer: msg_in_event] Stop mode enabled. Rejecting new events.
Dec  2 18:03:26 31861dd23a4c 2021-12-02 18:03:26,798 kytos.core.buffers:INFO buffers:46:  [buffer: msg_out_event] Stop mode enabled. Rejecting new events.
Dec  2 18:03:26 31861dd23a4c 2021-12-02 18:03:26,799 kytos.core.buffers:INFO buffers:46:  [buffer: app_event] Stop mode enabled. Rejecting new events.
``` 

As you can see, most of the cases, the behavior is like the one showed above: after the user request has been processed, the thread `save_status_on_storehouse` runs (so the event is created before Kytos buffers enters in Stop mode) and the request is persisted. 

The log messages `storehouse:INFO fs:83:  _load_from_file` and `storehouse:INFO fs:73:  _write_to_file ` were inserted to help troubleshooting:
```
root@31861dd23a4c:/src/kytos-storehouse# git diff
diff --git a/backends/fs.py b/backends/fs.py
index 8bae31b..8539fa6 100644
--- a/backends/fs.py
+++ b/backends/fs.py
@@ -70,6 +70,7 @@ class FileSystem(StoreBase):
         with lock:
             with open(filename, 'wb') as save_file:
                 pickle.dump(box, save_file)
+            log.info("_write_to_file done %s" % filename)

     def _load_from_file(self, filename):
         dotted_filename = str(filename).replace('/', '.')
@@ -79,6 +80,7 @@ class FileSystem(StoreBase):
             try:
                 with open(filename, 'rb') as load_file:
                     data = pickle.load(load_file)
+                log.info("_load_from_file done %s" % filename)
                 return data
             except pickle.PickleError:
                 return False
```

--- a/main.py
+++ b/main.py
@@ -687,19 +687,21 @@ class Main(KytosNApp):  # pylint: disable=too-many-public-methods
     #    if settings.DISPLAY_FULL_DUPLEX_LINKS:
     #        self.topology.add_link(host.id, interface.id)
 
-    # pylint: disable=unused-argument
     @listen_to('.*.network_status.updated')
-    def save_status_on_storehouse(self, event=None):
+    def handle_network_status_updated(self, event):
+        """Handle *.network_status.updated events, specially from of_lldp."""
+        content = event.content
+        log.info(f"Storing the administrative state of the"
+                 f" {content['attribute']} attribute to"
+                 f" {content['state']} in the interfaces"
+                 f" {content['interface_ids']}")
+        self.save_status_on_storehouse()
+
+    def save_status_on_storehouse(self):
         """Save the network administrative status using storehouse."""
         with self._lock:
             status = self._get_switches_dict()
             status['id'] = 'network_status'
-            if event:
-                content = event.content
-                log.info(f"Storing the administrative state of the"
-                         f" {content['attribute']} attribute to"
-                         f" {content['state']} in the interfaces"
-                         f" {content['interface_ids']}")
             status.update(self._get_links_dict())
             self.storehouse.save_status(status)
 

--- a/tests/unit/test_main.py
+++ b/tests/unit/test_main.py
@@ -906,6 +906,17 @@ class TestMain(TestCase):
         response = api.post(url)
         self.assertEqual(response.status_code, 404, response.data)
 
+    @patch('napps.kytos.topology.main.Main.save_status_on_storehouse')
+    def test_handle_network_status_updated(self, mock_save_status):
+        """Test handle_link_maintenance_start."""
+        event = MagicMock()
+        event.name = 'kytos/of_lldp.network_status.updated'
+        event.content = {'attribute': 'LLDP',
+                         'state': 'enabled',
+                         'interface_ids': '00:00:00:00:00:00:00:01:1'}
+        self.napp.handle_network_status_updated(event)
+        mock_save_status.assert_called_once()
+
     def test_get_link_metadata(self):
         """Test get_link_metadata."""
         mock_link = MagicMock(Link)

